csharp-intermediate-topics/DifferencesBetweenDynamicTypes/Tests/DynamicTypesUnitTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.ComponentModel;
+using DifferencesBetweenDynamicTypes;
+using System.Dynamic;
+using Microsoft.CSharp.RuntimeBinder;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Tests;
+
+public class DynamicTypesUnitTest
+{
+    private readonly ITestOutputHelper _testOutputHelper;
+
+    public DynamicTypesUnitTest(ITestOutputHelper testOutputHelper)
+    {
+        _testOutputHelper = testOutputHelper;
+    }
+
+    [Fact]
+    public void WhenAccessingNonExistentDynamicMember_ThenThrowError()
+    {
+        dynamic person = new
+        {
+            Name = "Test"
+        };
+
+        Assert.Throws<RuntimeBinderException>(() => person.Age);
+    }
+
+    [Fact]
+    public void WhenAssigningDifferentTypesToObjects_ThenReturnObjectTypes()
+    {
+        dynamic count = 1;
+
+        Assert.IsType<int>(count);
+    }
+
+    [Fact]
+    public void WhenCreateExpandoObjectMethodIsCalled_ThenReturnExpandoObject()
```

Comment:
The name of the test method doesn't match what it's doing. I think this is from the old code where we had a service class.